### ISSUE TITLE
Adding works that don't have pairtree prefixes/version files in the directory causes 500 error

### ISSUE THREAD

**Issue by mnaydan** [Princeton-CDH/ppa-django#538]
Discovered when trying to add uga1.32108002998303 to production, which is the first time I was adding a uga1 work to the database. 

@rlskoeser said that the directory for uga1.32108002998303 doesn't have the pairtree prefix and pairtree version files that are part of the spec and which the library we're using expects to be in place (HathiTrust doesn't use them). 

Potential solution is to create logic to create them (in the past we likely created them manually).

**Comment by rlskoeser** [comment]
was originally thinking we need to create these when we add files - but after encountering the error again recently, I think we should try a different approach: subclass the pairtree class we're using to make a hathi specific version, then catch the invalid pairtree exception - when it occurs, create the files that should be there to make it a valid pairtree.

**Comment by rlskoeser** [comment]
@laurejt If you want to check what files we get from HathiTrust rsync, you can manually run rsync commands from the prosody staging servers, they are both on the rsync allow list.

The default rsync configuration is in our main settings file, which is checked into version control:
https://github.com/Princeton-CDH/ppa-django/blob/main/ppa/settings.py#L245-L247

Here is the import code where we're generating a file list and using rsync to pull the content to import a new document:
https://github.com/Princeton-CDH/ppa-django/blob/main/ppa/archive/import_util.py#L246

The path list is generated by this method: https://github.com/Princeton-CDH/ppa-django/blob/main/ppa/archive/import_util.py#L196
You can see from the comments that it's adding the pairtree prefix and version files to the list; my recollection is that HathiTrust doesn't actually have those files on their server, but this would be good to confirm with a manual rsync; maybe there's something wrong with how those paths are being generated in the rsync file list.

Here's the git repo for the python pairtree library we're using: https://github.com/benosteen/pairtree - one thought is to extend the pairtree storage client and implement a version that handles that validation error (either create the files if they don't exist, or just ignore the error if we don't need the files - I don't remember if the prefix file is needed to generate the correct paths).

I recently discovered there's a much simpler pairtree library maintained by Ed Summers https://pypi.org/project/ptree/ - it only handles the path logic, so I suspect it doesn't do any validation, but switching to that would be a bigger refactor.

**Comment by rlskoeser** [comment]
@mnaydan if you would suggest a couple of hathi ids to try, I will run the hathi add script on the command line and see what happens. I think it may work; if it doesn't, we at least want to document the error.

**Comment by mnaydan** [comment]
@rlskoeser can I have a list of what prefixes are in the pairtree?

**Comment by rlskoeser** [comment]
yes!

here's what I see at the configured path on the staging server:
- aeu
- bc
- chi
- coo
- coo1
- dul1
- hvd
- iau
- ien
- inu
- loc
- mdp
- miun
- msu
- nc01
- njp
- nnc1
- nnc2
- nyp
- osu
- pst
- txu
- uc1
- uc2
- ucm
- udel
- uga1
- uiuc
- uiug
- uiuo
- umn
- uva
- wu
- yale
- yul

**Comment by mnaydan** [comment]
@laurejt @rlskoeser I tested the following IDs in the admin interface and they all failed to sync data :/ 

pur1.32754080978426
keio.10810522675
ucw.ark:/13960/t0wq0b01q

**Comment by rlskoeser** [comment]
@mnaydan thanks for confirming, that's helpful

**Comment by rlskoeser** [comment]
@mnaydan as I suspected (😞 😬 ) the problem is also permissions. Ownership of NFS top-level vs data within it. I'm going to try updating the deploy to revise how we set permissions on the pairtree top-level folder (again).

**Comment by rlskoeser** [comment]
@mnaydan @laurejt  I modified the ansible deploy to change file ownership to www-data/conan, and manually confirmed I'm able to create a file in the hathi data directory as the www-data user.

@mnaydan please try testing again when you get the chance.

**Comment by mnaydan** [comment]
Success!! I tested with the above list and it successfully added pur1.32754080978426 and keio.10810522675 and skipped ucw.ark:/13960/t0wq0b01q (I guess that's the one you added). I tried two additional ones to be safe: ia.ark:/13960/s2n30d0sbj3 and miua.1185748.0001.001.

@rlskoeser did you want to check a few via script as you mentioned above? I would use:
ucbk.ark:/28722/h2z892g9f
ppt.ssfcbz201710000324
asu1.a15017253471
